Dockerfile.s3

```diff
@@ -0,0 +1,129 @@
+# base: Stage which installs necessary runtime dependencies (OS packages, java,...)
+FROM python:3.11.5-slim-buster@sha256:9f35f3a6420693c209c11bba63dcf103d88e47ebe0b205336b5168c122967edf as base
+ARG TARGETARCH
+
+# set workdir
+RUN mkdir -p /opt/code/localstack
+WORKDIR /opt/code/localstack/
+
+# Install runtime OS package dependencies
+RUN --mount=type=cache,target=/var/cache/apt \
+    apt-get update && \
+        # Install dependencies to add additional repos
+        apt-get install -y --no-install-recommends \
+            # Runtime packages (groff-base is necessary for AWS CLI help)
+             ca-certificates curl make openssl
+# TODO: add this if we need the DNS server: iputils-ping iproute2
+
+SHELL [ "/bin/bash", "-c" ]
+
+# create localstack user and filesystem hierarchy, perform some permission fixes
+RUN chmod 777 . && \
+    useradd -ms /bin/bash localstack && \
+    mkdir -p /var/lib/localstack && \
+    chmod -R 777 /var/lib/localstack && \
+    mkdir -p /usr/lib/localstack && \
+    mkdir /tmp/localstack && \
+    chmod -R 777 /tmp/localstack && \
+    touch /tmp/localstack/.marker
+
+# install basic (global) tools to final image
+RUN --mount=type=cache,target=/root/.cache \
+    pip install --no-cache-dir --upgrade virtualenv
```

Comment:
Is this necessary in the image? This could be neglected, or moved down to the installation of the dependencies.

Reply:
Oh, good question. This might have been kept from the base `Dockerfile`, I will check. But quick question, why would this not be necessary in this image but in the default `localstack` image?

Reply:
To be honest, I think the installation of `virtualenv` is a remnant from older times. I don't think we use it anywhere anymore (I think we use the builtin `venv` module of Python 3 everywhere).

Reply:
Oh okay, great, I'll remove 👍